### PR TITLE
[wicket/wicketd] Retrieve SpComponentList

### PR DESCRIPTION
**Pull request by andrewjstone**
Add SpComponentInfo to RackV1Inventory by pulling it from MGS as necessary. We only pull from MGS when the state of the SP has changed or we do not yet have the component info. This is because the info is static except for when the SP is upgraded.

We then expose this information via the wicketd-client with the appropriately derived traits and plumb it up through wicket.

### CODE REVIEW

File: wicket/src/wicketd.rs [hunk 1]
```diff
@@ -113,11 +113,11 @@ async fn poll_inventory(
...
                     let new_inventory = val.into_inner();
-                    // TODO: We need the derive enhancements for progenitor to implement PartialEq/Eq
-                    //if new_inventory != inventory {
-                    //  inventory = new_inventory;
-                    //}
-                    let _ = tx.send(new_inventory).await;
+                    if new_inventory != inventory {
+                        inventory = new_inventory;
+                        let _ = tx.send(inventory.clone()).await;
+                        debug!(log, "No change to inventory");
```
Comment (jgallagher) [inline wicket/src/wicketd.rs:119]: Should this log message be in an `else` block?

File: wicketd-client/src/lib.rs [hunk 2]
```diff
@@ -4,6 +4,27 @@
...
+    derives = [schemars::JsonSchema],
+    patch =
+        {
+        SpIdentifier = { derives = [Copy, PartialEq, Eq, PartialOrd, Ord] },
+        SpState = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpComponentInfo= { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpIgnition= { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpInventory = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        RackV1Inventory = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+    }
```
Comment (jgallagher) [inline wicketd-client/src/lib.rs:29]: Sweet!

**Review by jgallagher — APPROVED** [review]
Looks great!

File: wicket/src/wicketd.rs (hunk 1 above)
Comment (andrewjstone) [reply wicket/src/wicketd.rs:119]: oh, what the heck hahaha. Good catch!

File: wicketd-client/src/lib.rs (hunk 2 above)
Comment (andrewjstone) [reply wicketd-client/src/lib.rs:29]: @ahl did an awesome job here. It just worked!

**Pull request merged**